### ISSUE TITLE
"Is defined but not used" code diagnostic warning does not account for use in glue string

### ISSUE THREAD

**Issue by fh-jsnider** [rstudio/rstudio#5270]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1335
    OS Version      : macOS 10.13.6
    R Version       : 3.6.1

### Steps to reproduce the problem
```
  FILE_LOC <- "~/Desktop/"
  
  write.csv(mtcars, file = glue::glue("{file_loc}/mt.cars.csv"))
```
### Describe the problem in detail
If in options, "Diagnostics/ Warn if variable is defined but not used" is checked, the above will produce an incorrect warning.
### Describe the behavior you expected
No warning is issued.

**Comment by ronblum** [comment]
@fh-jsnider Thank you for filing the issue. I notice that the variable instances have different casing—uppercase  `FILE_LOC` and lowercase `file_loc`. Could that be the problem? I ran the code above, and got the error
```
Error in eval(parse(text = text, keep.source = FALSE), envir) : 
  object 'file_loc' not found
```
But when changing to matching case, the file got created as expected. Also, what is the warning that you get?

**Comment by fh-jsnider** [comment]
No, that's not it (and sorry, they both should have been the same case!

Here's an example of what I mean.

![image](https://user-images.githubusercontent.com/50179237/63593930-2134b200-c583-11e9-94af-d257ecee796d.png)

**Comment by ronblum** [comment]
@fh-jsnider Thank you! I see—the code diagnostics doesn't recognize that `glue` interprets the string, instead treating it like any other string literal. I'll mark this as enhancement—to recognize a string's code that will be interpreted by `glue`—to review as we continue development of RStudio.

```
used_no <- function() {
  FILE_LOC <- "~/Desktop/"
  return(1)
}

used_plain <- function() {
  FILE_LOC <- "~/Desktop/"
  print(FILE_LOC)
}

used_quotes <- function() {
  FILE_LOC <- "~/Desktop/"
  print("{FILE_LOC}")
}

used_glue <- function() {
  FILE_LOC <- "~/Desktop/"
  print(glue::glue("{FILE_LOC}/mt.cars.csv"))
}
```
<img width="427" alt="glue" src="https://user-images.githubusercontent.com/31009952/63600327-5d6f0f00-c591-11e9-8909-5b6ce943e274.png">

Confirmed in RStudio Desktop 1.2.1571 on MacOS 10.15 (Catalina) and RStudio Server 1.2.1568 on Red Hat 8.

**Comment by ronblum** [comment]
Another example from #3821
```
library(glue)
greeting <- function() {
   name <- "xan"
   msg <- glue("hello {name}")
   msg
}
greeting()
```

**Comment by dfalty** [comment]
Verified fixed 1.3.867
